pcsx2/gui/FrameForGS.cpp

```diff
@@ -642,21 +641,19 @@ void GSFrame::OnUpdateTitle( wxTimerEvent& evt )
 	out << std::fixed << std::setprecision(2) << fps;
 	OSDmonitor(Color_StrongGreen, "FPS:", out.str());
 
-	int new_width = 0, new_height = 0;
+	const u64& smode2 = *(u64*)PS2GS_BASE(GS_SMODE2);
+	int new_width = 0, new_height = 0, interlaced = !g_Conf->GSWindow.DisableScalingInterlaceCompensation && (smode2 & 2) && (smode2 & 2);
 	AspectRatioType Aspect_Ratio = g_Conf->GSWindow.AspectRatio;
 	if (GSgetImageSize != nullptr && (Aspect_Ratio == AspectRatio_Frame || (Aspect_Ratio != AspectRatio_Stretch && g_Conf->GSWindow.ScalingType != ScalingType_Fit))) {
 		GSgetImageSize(&new_width, &new_height);
-		if (new_width > 0 && new_height > 0)
+		if (new_width > 0 && new_height > 0 && (image_width != new_width || (image_height != new_height && !interlaced) || (image_height == new_height && interlaced)))
 		{
 			if (GSPanel* gsPanel = GetViewport())
 			{
-				if (image_width != new_width || image_height != new_height)
-				{
 					image_width = new_width;
-					image_height = new_height;
+					image_height = interlaced ? new_height * 2 : new_height;
```

Comment:
``new_height * ( 1 + interlaced)`` could have been nice, but you can't do it if you change it to bool though.

Reply:
> but you can't do it if you change it to bool though.

Yes, you can. :)

It could also have been reduced further to image_height *= 1 + interlaced;

Reply:
Cool, didn't know about it. I thought it might require an integer cast, but casting a boolean to integer might look awkward. (Not even sure if upcasting like that is even possible)

Or is the bool just taken as an integer without any sort of cast?

Reply:
> Or is the bool just taken as an integer without any sort of cast?

Yes, as far as I know it still is. Though I agree that it can look awkward.